### ISSUE TITLE
query_bin_or_ascii_float_list() ValueError: could not convert string to float

### ISSUE THREAD

**Issue by tomasznm** [Rohde-Schwarz/RsInstrument#18]
Issue Description:

I am using R&S RTB2004 on Ubuntu 22.04 and set the scope correctly.

However, what appears on the last chunk transfer has some problems with data parsing on:

my_data_trace = rtb.query_bin_or_ascii_float_list("FORM ASC;:CHAN:DATA?")

I get this error:

Exception has occurred: ValueError
could not convert string to float: '7.32422E-01\n\x00\x00'
  File "/work/scopes/RsInstrument_Acquisition.py", line 139, in <module>
    trace = rtb.query_bin_or_ascii_float_list("FORM ASC;:CHAN:DATA?")
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: could not convert string to float: '7.32422E-01\n\x00\x00'

It looks like this: "**\n\x00\x00**" may be the problem.

I have the logging enabled:

DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(32)') -> 0
DEBUG:pyvisa:viWrite(2, b'*ESR?', 5, 'c_uint(5)') -> 0
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_1024 object at 0x7cfc76751150>, 1024, 'c_uint(2)') -> 0
DEBUG:pyvisa:viWrite(2, b'*STB?', 5, 'c_uint(5)') -> 0
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_1024 object at 0x7cfc7679e950>, 1024, 'c_uint(2)') -> 0
DEBUG:pyvisa:viWrite(2, b'', 0, 'c_uint(0)') -> 0
DEBUG:pyvisa:viWrite(2, b'*STB?', 5, 'c_uint(5)') -> 0
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_1024 object at 0x7cfc76751150>, 1024, 'c_uint(2)') -> 0
DEBUG:pyvisa:viWrite(2, b'FORM ASC;:CHAN:DATA?', 20, 'c_uint(20)') -> 0
DEBUG:pyvisa:USB0::0x0AAD::0x01D6::112025::0::INSTR - reading 1 bytes (last status None)
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_1 object at 0x7cfc76751150>, 1, 'c_uint(1)') -> 1073676294
DEBUG:pyvisa:viReadSTB(2, 'c_ushort(0)') -> 0
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_1024 object at 0x7cfc76751150>, 1024, 'c_uint(1024)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_65536 object at 0x7cfc76751150>, 65536, 'c_uint(65536)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc7679e950>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc7679e950>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc76751150>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc7679e950>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc76751150>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc7679e950>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc76751150>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc7679e950>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc76751150>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc7679e950>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc76751150>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc7679e950>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc76751150>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc7679e950>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc76751150>, 100000, 'c_uint(100000)') -> 1073676294
DEBUG:pyvisa:viRead(2, <ctypes.c_char_Array_100000 object at 0x7cfc7679e950>, 100000, 'c_uint(192)') -> 0

It transferred a lot of data, and the last chunk only had what appeared to be 192 items.

Would there be any hint on how to fix this?

**Comment by tomasznm** [comment]
RsInstrument driver version: 1.82.1.106

**Comment by Miloslav-RS** [comment]
Hi Tomasz, sorry, I overlooked the issue. Is it still actual? Seems like the instrument returns some strange characters. It might help switching to binary transfer, especially if you transfer a lot of characters.

**Comment by tomasznm** [comment]
Yes, it was last year. I moved to a different oscilloscope brand that allowed me to pump the data without problems. Thanks.
